pgjdbc/src/main/java/org/postgresql/PGEnvironment.java

```diff
@@ -47,7 +127,7 @@ public enum PGEnvironment {
    */
   PGSERVICEFILE(
       "PGSERVICEFILE",
-      "pg_service.conf",
+      null,
```

Comment:
Is this a backward compatible change?

Reply:
There is no one constant default value for given parameter.
The default value is operating system specific.
**Yes, functionality is backward compatible (the same).**

**Can you recommend proper approach?**
This enum is not meant to be used outside pgjdbc driver.
There should not be any promises assumed regarding this enum outside pgjdbc driver.
At the same time I can not remove keyword `public`.
How could we avoid "backward compatible" questions which are relevant only because of technical restrictions?

Reply:
>This enum is not meant to be used outside pgjdbc driver.

The enum is public, it is located in a public `org.postgresql` package, nothing in its javadoc says anything regarding "for pgjdbc internal use only" and so on.

There might be usages of the enum outside of the driver.

---

>How could we avoid "backward compatible" questions which are relevant only because of technical restrictions?

I do not understand the question.
I think the only case we can merge a PR is if the PR keeps full backward compatibility or if the breakage is minimal.

If there's a significant breakage of the backward compatibility, there's very little we can do.

Reply:
> ... is if the PR keeps full backward compatibility or if the breakage is minimal.

In my estimation, the real impact is close to zero.
The old value is incorrect and there is no common constant default value for this parameter across all operating systems.
If someone has adopted this default value and has tested their system, they must have realized that this value is wrong

Reply:
It does not have to work across all the OS. It might work for a single os, so changing default might break user's workfows.

We focus on backward compatibility, so breaking things like "property no longer used" or "enum entry no longer exists" makes it much harder to merge the change.

Could you please double-check if the PR maintains backward compatibility?

Reply:
The "old" default value is incorrect for all major operating systems: Linux, Windows, and MacOS.
A bit more correct value would be: `.pg_service.conf`
Correct value would be: `${HOME}/.pg_service.conf` for Linux/MacOS and `%APPDATA%\postgresql\.pg_service.conf` for Windows

I disagree that maintaining this incorrect default value is necessary, but since it seems to be a blocking issue, I have reverted code to keep the incorrect "old" default value.

Additionally, I have marked the `getDefaultValue()` function as deprecated.
This serves as a reminder that in the next major version, both the function and the incorrect default value may be removed from the code.